examples/apps/color_finder.py

```diff
@@ -0,0 +1,475 @@
+        WM = 10.0*32/self.grid_width  # WM and HM are multipliers that adjust the amount to turn
+        HM = 10.0*24/self.grid_height # based on the current dimensions of self.pixel_matrix.
+        self.robot.set_center_backpack_lights(map_color_to_light[self.color_to_find])
+
+        # 'fov' stands for 'field of view'. This is the angle amount
+        # that Cozmo can see to the edges of his camera view.
+        amount_move_head = radians(self.robot.camera.config.fov_y.radians*(120-y*HM)/240)
+        amount_rotate = radians(self.robot.camera.config.fov_x.radians*(160-x*WM)/320)
+        if self.moved_too_far_from_center(amount_move_head,amount_rotate):
+            self.state='found_color'
+        if self.state != 'driving':
+            self.look_at_it(amount_move_head,amount_rotate)
+        else:
+            self.drive_at_it()
+
+    def moved_too_far_from_center(self,amount_move_head,amount_rotate):
+        ''' Decides whether the center of the blob is too far from the approximate center of Cozmo's view.'''
+        too_far_vertical = abs_val(amount_move_head)>self.robot.camera.config.fov_y/8
+        too_far_horizontal = abs_val(amount_rotate)>self.robot.camera.config.fov_x/8
+        return (too_far_vertical or too_far_horizontal)
+
+    def look_at_it(self,amount_move_head,amount_rotate):
+        ''' Calls actions that tilt Cozmo's head and rotate his body toward the color.'''
+        self.robot.abort_all_actions()
+        new_head_angle = self.robot.head_angle + amount_move_head
+        self.tilt_head_action = self.robot.set_head_angle(new_head_angle, in_parallel=True)
+        self.rotate_action = self.robot.turn_in_place(amount_rotate, in_parallel=True)
+        if self.state=='found_color':
+            self.amount_turned_recently += abs_val(amount_move_head) + abs_val(amount_rotate)
+
+    def drive_at_it(self):
+        ''' Drives straight once prior actions have been cancelled.'''
+        if self.tilt_head_action.is_running:
+            self.tilt_head_action.abort()
+        if self.rotate_action.is_running:
+            self.rotate_action.abort()
+        if self.drive_action == None or not self.drive_action.is_running:
+            self.drive_action = self.robot.drive_straight(distance_mm(500), speed_mmps(100), should_play_anim=False, in_parallel=True)
+
+    def turn_on_cubes(self):
+        ''' Illuminates the two cubes that control self.color_to_find and set the Tkviewer display.'''
+        self.color_selector_cube = self.robot.world.get_light_cube(cozmo.objects.LightCube1Id)
+        self.color_selector_cube.set_lights(map_color_to_light[self.color_to_find])
+        self.grid_cube = self.robot.world.get_light_cube(cozmo.objects.LightCube2Id)
+        self.grid_cube.set_lights(cozmo.lights.white_light.flash())
+
+    def setup(self):
+        self.robot.set_head_angle(radians(0))
+        self.robot.drive_straight(distance_mm(50), speed_mmps(100), should_play_anim=False, in_parallel=True)
```

Comment:
sometimes he would spot a color immediately while on the charging deck and start trying to turn immediately, scraping the wheels against the charger.  

Replacing this with 
``cozmo.robot.Robot.drive_off_charger_on_connect = True``